### PR TITLE
Added http-api with authenticator

### PR DESCRIPTION
**Pull request by Jess3Jane**
- Added http-api
  - Added authenticator middeware to http-api
  - Removed old qaul-http api
  - Changed identity to be Copy/Clone
  - Added user stub
  - Cleaned up a few imports

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -11,6 +11,9 @@ members = [
     "libqaul/service/files",
     "libqaul/service/core",
 
+	# qaul.net http-api
+	"libqaul/http-api",
```
Comment (spacekookie) [inline Cargo.toml]: Can you please use spaces for indentation

File: libqaul/http-api/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+[package]
+name = "http-api"
```
Comment (spacekookie) [inline libqaul/http-api/Cargo.toml]: The crate name should probably be more descriptive like `qaul-http` :sweat_smile:

File: libqaul/http-api/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    }
+}
+
+struct QaulCore{
+    qaul: Arc<Qaul>,
+}
+
+impl QaulCore {
+    pub fn new(qaul: Arc<Qaul>) -> QaulCore {
+        QaulCore{
```
Comment (spacekookie) [inline libqaul/http-api/src/lib.rs]: Formatting is a bit weird here. Also maybe just use `Self`

**Review by spacekookie — APPROVED** [review]
There's some small formatting changes, but otherwise this looks very good.

I assume that part of what the authenticator does here will be able to be moved to the `auth` module in `libqaul` at some point, but having it here for now is totally cool to get started

File: libqaul/http-api/src/lib.rs (hunk 3 above)
Comment (Jess3Jane) [reply libqaul/http-api/src/lib.rs]: I changed things to use `Self` and reformatted a bit, if you have any other formatting concerns let me know

**Comment by Jess3Jane** [comment]
I also added a commit that derives `Clone` on `Qaul` to allow references to be shared across thread boundaries as discussed on the Zulip

I'm still not overly sold on this way of doing it but we need some way to share `&Qaul` across thread boundaries

**Comment by spacekookie** [comment]
Yea, agreed. It feels like more of a hack for now. What we could possibly do is just do a type alias for `Qaul = Arc<ActualQaul>` or something. But we can bikeshed this later too.

**Pull request merged**